packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -34,47 +35,50 @@ class _UserAvatarState extends State<NeonUserAvatar> {
   }
 
   @override
-  Widget build(final BuildContext context) {
-    final isDark = Theme.of(context).brightness == Brightness.dark;
-    final size = (widget.size * MediaQuery.of(context).devicePixelRatio).toInt();
-    return Stack(
-      alignment: Alignment.center,
-      children: [
-        CircleAvatar(
-          radius: widget.size / 2,
-          child: ClipOval(
-            child: NeonCachedApiImage(
-              account: widget.account,
-              cacheKey: 'avatar-${widget.username}-${isDark ? 'dark' : 'light'}$size',
-              download: () async {
-                if (isDark) {
-                  return widget.account.client.core.getDarkAvatar(
-                    userId: widget.username,
-                    size: size,
-                  );
-                } else {
-                  return widget.account.client.core.getAvatar(
-                    userId: widget.username,
-                    size: size,
-                  );
-                }
-              },
-            ),
-          ),
-        ),
-        if (widget.showStatus) ...[
-          ResultBuilder<NextcloudUserStatusPublicStatus?>(
-            stream: _userStatusBloc.statuses.map((final statuses) => statuses[widget.username]),
-            builder: _userStatusIconBuilder,
-          ),
-        ],
-      ],
-    );
-  }
+  Widget build(final BuildContext context) => LayoutBuilder(
+        builder: (final context, final constraints) {
+          final isDark = Theme.of(context).brightness == Brightness.dark;
+          size = constraints.constrain(Size.square(widget.size)).shortestSide;
+          final pixelSize = (size * MediaQuery.of(context).devicePixelRatio).toInt();
```

Comment:
Why do we need this? can't we use size directly?

Reply:
The flutter size is not the same size as physical pixels. To get an image with the correct resolution for the display we need to convert the flutter size into the physical pixel size.

Reply:
>   /// The number of device pixels for each logical pixel. This number might not
  /// be a power of two. Indeed, it might not even be an integer. For example,
  /// the Nexus 6 has a device pixel ratio of 3.5.

Reply:
Ah yeah makes sense